// @iframe-resizer/jquery/index.esm.js
/*!
 *  @preserve
 *  
 *  @module      iframe-resizer/jquery 5.5.8 (esm) - 2026-01-26
 *
 *  @license     GPL-3.0 for non-commercial use only.
 *               For commercial use, you must purchase a license from
 *               https://iframe-resizer.com/pricing
 * 
 *  @description Keep same and cross domain iFrames sized to their content 
 *
 *  @author      David J. Bradshaw <info@iframe-resizer.com>
 * 
 *  @see         {@link https://iframe-resizer.com}
 * 
 *  @copyright  (c) 2013 - 2026, David J. Bradshaw. All rights reserved.
 */


import connectResizer from '@iframe-resizer/core';
import acg from 'auto-console-group';

const LABEL = 'iframeResizer';
const NEW_LINE = '\n';
const PARENT = 'parent';

const STRING = 'string';

const deprecate = (advise) =>
  (type, change = 'renamed to') =>
  (old, replacement, info = '', iframeId = '') =>
    advise(
      iframeId,
      `<rb>Deprecated ${type}(${old.replace('()', '')})</>\n\nThe <b>${old}</> ${type.toLowerCase()} has been ${change} <b>${replacement}</>. ${info}Use of the old ${type.toLowerCase()} will be removed in a future version of <i>iframe-resizer</>.`,
    );

const isString = (value) => typeof value === STRING;

const id = (x) => x;

const esModuleInterop = (mod) =>
  // eslint-disable-next-line no-underscore-dangle
  mod?.__esModule ? mod.default : mod;

/* eslint-disable no-useless-escape */
/* eslint-disable security/detect-non-literal-regexp */


const TAGS = {
  br: '\n',
  rb: '\u001B[31;1m', // red bold
  bb: '\u001B[34;1m', // blue bold
  b: '\u001B[1m', // bold
  i: '\u001B[3m', // italic
  u: '\u001B[4m', // underline
  '/': '\u001B[m', // reset
};

const keys = Object.keys(TAGS);
const tags = new RegExp(`<(${keys.join('|')})>`, 'gi');
const lookup = (_, tag) => TAGS[tag] ?? '';
const encode = (s) => s.replace(tags, lookup);

const filter = (s) =>
  s.replaceAll('<br>', NEW_LINE).replaceAll(/<\/?[^>]+>/gi, '');

const createFormatAdvise = (formatLogMessage) => (message) =>
  formatLogMessage(
    isString(message)
      ? window.chrome
        ? encode(message)
        : filter(message)
      : message,
  );

/* eslint-enable security/detect-non-literal-regexp */
/* eslint-enable no-useless-escape */

const settings = {};

// Deal with UMD not converting default exports to named exports
const createConsoleGroup = esModuleInterop(acg);

const parent = createConsoleGroup({
  expand: false,
  label: PARENT,
});

const output =
  (type) =>
  (iframeId, ...args) =>
    settings[iframeId]
      ? settings[iframeId].console[type](...args)
      : parent[type](...args);
const warn = output('warn');

const formatLogMsg =
  (iframeId) =>
  (...args) =>
    [`${LABEL}(${iframeId})`, ...args].join(' ');

const formatAdvise = createFormatAdvise(id);
const advise = (iframeId, ...args) =>
  settings[iframeId]
    ? settings[iframeId].console.warn(...args.map(formatAdvise))
    : queueMicrotask(() => {
        const localFormatAdvise = createFormatAdvise(formatLogMsg(iframeId));
        // eslint-disable-next-line no-console
        console?.warn(...args.map(localFormatAdvise));
      });

const deprecateAdvise = deprecate(advise);
const deprecateMethod = deprecateAdvise('Method');

switch (true) {
  case window.jQuery === undefined:
    warn('', 'Unable to bind to jQuery, it is not available.');
    break

  case !window.jQuery.fn:
    warn('', 'Unable to bind to jQuery, it is not fully loaded.');
    break

  case window.jQuery.fn.iframeResize:
    warn('', 'iframeResize is already assigned to jQuery.fn.');
    break

  default:
    window.jQuery.fn.iframeResize = function (options) {
      const connectWithOptions = connectResizer(options);
      const init = (i, el) => connectWithOptions(el);

      return this.filter('iframe').each(init).end()
    };

    window.jQuery.fn.iFrameResize = function (options) {
      deprecateMethod('iFrameResize()', 'iframeResize()', '', 'jQuery');

      return this.iframeResize(options)
    };
}
